main.py

```diff
@@ -919,9 +919,10 @@ def get_redirect_route(
 
 # Redirect all routes handled using angular router to the oppia root page.
 for page in constants.PAGES_REGISTERED_WITH_FRONTEND.values():
-    URLS.append(
-        get_redirect_route(
-            r'/%s' % page['ROUTE'], oppia_root.OppiaRootPage))
+    if not 'MANUALLY_REGISTERED_WITH_BACKEND' in page:
```

Comment:
Can you explain what's `MANUALLY_REGISTERED_WITH_BACKEND` and why it is introduced? :)

Reply:
Some URLs contain URL fragments and they have different syntax for angular router and backend.
eg.
Url for backend: /learn/<classroom_url_fragment>
url for angular router: /learn/:classroom_url_fragment

So, I have used the second case in constant.ts (so, it works directly with angular router).

And manually registered these routes with fragments in the backend.
That's why I am excluding these routes in this loop.